packages/flutter/lib/src/widgets/tween_animation_builder.dart

```diff
@@ -0,0 +1,249 @@
+///             });
+///           },
+///         );
+///       },
+///       child: Icon(Icons.aspect_ratio),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// ## Relationship to [ImplicitlyAnimatedWidget]s and [AnimatedWidget]s
+///
+/// The [ImplicitlyAnimatedWidget] has many subclasses that provide animated
+/// versions of regular widgets. These subclasses (like [AnimatedOpacity],
+/// [AnimatedContainer], [AnimatedSize], etc.) animate changes in their
+/// properties smoothly and they are easier to use than this general-purpose
+/// builder. However, [TweenAnimationBuilder] (which itself is a subclass of
+/// [ImplicitlyAnimatedWidget]) is handy for animating any widget property to a
+/// given target value even when the framework (or third-party widget library)
+/// doesn't ship with an animated version of that widget.
+///
+/// Those [ImplicitlyAnimatedWidget]s (including this [TweenAnimationBuilder])
+/// all manage an internal [AnimationController] to drive the animation. If you
+/// want more control over the animation than just setting a target value,
+/// [duration], and [curve], have a look at (subclasses of) [AnimatedWidget]s.
+/// For those, you have to manually manage an [AnimationController] giving you
+/// full control over the animation. An example of an [AnimatedWidget] is the
+/// [AnimatedBuilder], which can be used similarly to this
+/// [TweenAnimationBuilder], but unlike the latter it is powered by a
+/// developer-managed [AnimationController].
+class TweenAnimationBuilder<T> extends ImplicitlyAnimatedWidget {
+  /// Creates a [TweenAnimationBuilder].
+  ///
+  /// The properties [tween], [duration], and [builder] are required. The values
+  /// for [tween], [curve], and [builder] must not be null.
+  ///
+  /// The [TweenAnimationBuilder] takes full ownership of the provided [tween]
+  /// instance and mutates it. Once a [Tween] has been passed to a
+  /// [TweenAnimationBuilder], its properties should not be accessed or changed
+  /// anymore to avoid interference with the [TweenAnimationBuilder].
+  const TweenAnimationBuilder({
+    Key key,
+    @required this.tween,
+    @required Duration duration,
+    Curve curve = Curves.linear,
+    @required this.builder,
+    this.onEnd,
+    this.child,
+  }) : assert(tween != null),
+       assert(curve != null),
+       assert(builder != null),
+       super(key: key, duration: duration, curve: curve);
+
+  /// Defines the target value for the animation.
+  ///
+  /// When the widget first builds, the animation runs from [Tween.begin] to
+  /// [Tween.end], if [Tween.begin] is non-null. A new animation can be
+  /// triggered at anytime by providing a new [Tween] with a new [Tween.end]
+  /// value. The new animation runs from the current animation value (which may
+  /// be [Tween.end] of the old [tween], if that animation completed) to
+  /// [Tween.end] of the new [tween]. The [Tween.begin] value is ignored except
```

Comment:
Is there any special case when a new Tween.end is < the original Tween.begin? How does that affect the animation - and the direction as well? 
It seems that all of the examples and tests illustrate that Tween.begin < Tween.end.

Reply:
Tweens in general don't make assumptions about how begin and end relate to each other. end can be smaller then begin and it shouldn't be a problem to run a tween forward from 100 to 10.